File: week_1/06_find_not_repeating_first_character.js
input = "abacabade"

const alphabet_count = input.split('').reduce((acc, value) => {
    return {
        ...acc,
        [value]: acc[value] ? acc[value] + 1 : 1
    }
},{})

const notRepeatedAlphabets = Object.entries(alphabet_count).filter(x => x[1] === 1).map(x => x[0])

let answer
for (const alphabet of input) {
    if (notRepeatedAlphabets.includes(alphabet)) {
        answer = alphabet
        break
    }
}

console.log(answer)

File: week_1/06_find_not_repeating_first_character.py
input = "abacabade"


# def find_not_repeating_character(string):
#     # 이 부분을 채워보세요!
#     alphabet_count = {}
#     for alphabet in input:
#         if alphabet in alphabet_count.keys():
#             alphabet_count[alphabet] += 1
#         else:
#             alphabet_count[alphabet]
#     return "_"


# result = find_not_repeating_character(input)
# print(result)

alphabet_count = {}
for alphabet in input:
    if alphabet in alphabet_count.keys():
        alphabet_count[alphabet] += 1
    else:
        alphabet_count[alphabet] = 1

print(alphabet_count.items())

# 미완. 현재 익숙한 자바스크립트로 풀기로